src/Renderer/Camera.js

```diff
@@ -87,9 +47,60 @@ Camera.prototype.update = function update(width, height) {
     // horizontalPreSSE = this.width / (2.0 * Math.tan(verticalFOV * 0.5) * this.width / this.height)
     //                  = this.height / 2.0 * Math.tan(verticalFOV * 0.5)
     //                  = verticalPreSSE
-    this._preSSE = verticalPreSSE;
+    camera._preSSE = verticalPreSSE;
+}
+
+function Camera(crs, width, height, options = {}) {
+    Object.defineProperty(this, 'crs', { get: () => crs });
+
+    this.camera3D = options.camera ? options.camera : new THREE.PerspectiveCamera(30, width / height);
+
+    this._viewMatrix = new THREE.Matrix4();
+    this._viewMatrixNeedsUpdate = true;
+    this.resize(width, height);
+
+    this._preSSE = Infinity;
+
+    if (this.camera3D.isPerspectiveCamera) {
+        let fov = this.camera3D.fov;
+        Object.defineProperty(this.camera3D, 'fov', {
+            get: () => fov,
+            set: (newFov) => {
+                fov = newFov;
+                updatePreSse(this, this.height, fov);
+            },
+        });
+    }
+}
+
+Camera.prototype.resize = function resize(width, height) {
+    this.width = width;
```

Comment:
This would duplicate things, but maybe you could also declare those variables in the constructor too ?

Reply:
ok